clang/bindings/python/tests/cindex/test_type.py

```diff
@@ -357,6 +359,50 @@ def test_is_restrict_qualified(self):
         self.assertTrue(i.type.is_restrict_qualified())
         self.assertFalse(j.type.is_restrict_qualified())
 
+    def test_get_result(self):
+        tu = get_tu("void foo(); int bar(char, short);")
+        foo = get_cursor(tu, "foo")
+        bar = get_cursor(tu, "bar")
+        self.assertEqual(foo.type.get_result().spelling, "void")
+        self.assertEqual(bar.type.get_result().spelling, "int")
+
+    def test_get_class_type(self):
+        tu = get_tu(
+            """
+class myClass
+{
+   char *myAttr;
+};
+
+char *myClass::*pMyAttr = &myClass::myAttr;
+""",
+            lang="cpp",
+        )
+        cur = get_cursor(tu, "pMyAttr")
+        self.assertEqual(cur.type.get_class_type().spelling, "myClass")
+
+    def test_get_named_type(self):
+        tu = get_tu("using char_alias = char; char_alias xyz;", lang="cpp")
+        cur = get_cursor(tu, "xyz")
+        self.assertEqual(cur.type.kind, TypeKind.ELABORATED)
+        self.assertEqual(cur.type.get_named_type().spelling, "char_alias")
```

Comment:
I have genuinely no idea what the point of `get_named_type` is: `cur.type.get_named_type()` is always equal to `cur.type` as far as I can tell.

Reply:
As discussed in https://github.com/llvm/llvm-project/pull/108769/files#r1772023859, I think we should be forward-looking here, and not rely on a bogus `ElaboratedType` node in AST for type aliases.

Reply:
Does that mean I should just remove the `self.assertEqual(cur.type.kind, TypeKind.ELABORATED)`, or what exactly?

Reply:
After reading through the discussion again, wasn't that only related to enums? Aren't type aliases elaborated types?

Reply:
This assert will stop passing soon, hopefully, so yeah, you should remove it. It seems that there's enough substance in the test even without the assert, so I think simply removing it should be sufficient.

Reply:
> After reading through the discussion again, wasn't that only related to enums? Aren't type aliases elaborated types?

I don't believe they are. Even more, elaborated types are not types but type specifiers, i.e. a way to name/reference a type, as opposed to being a property of declaration (of a type).